### PR TITLE
feat(apig): add endpoint service permissions sdk support

### PR DESCRIPTION
**Pull request by jinyangyang222**
**What this PR does / why we need it**:
add endpoint service permissions sdk support
**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format,
  will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: openstack/apigw/dedicated/v2/whitelist/requests.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+	var r createResp
+	_, err = c.Post(addURL(c, opts.InstanceId), b, &r, &golangsdk.RequestOpts{
+		MoreHeaders: requestOpts.MoreHeaders,
+	})
+	return r.Permissions, err
+}
+
+// ListWhiteList is a method used to query the whitelists list of endpoint service with given parameters.
+func ListWhiteList(c *golangsdk.ServiceClient, opts ListOpts) ([]EndpointPermission, error) {
```
Comment (Lance52259) [inline openstack/apigw/dedicated/v2/whitelist/requests.go:47]:
```suggestion
func List(c *golangsdk.ServiceClient, opts ListOpts) ([]EndpointPermission, error) {
```

File: openstack/apigw/dedicated/v2/whitelist/requests.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+	}
+
+	var r createResp
+	_, err = c.Post(addURL(c, opts.InstanceId), b, &r, &golangsdk.RequestOpts{
+		MoreHeaders: requestOpts.MoreHeaders,
+	})
+	return r.Permissions, err
+}
+
+// ListWhiteList is a method used to query the whitelists list of endpoint service with given parameters.
```
Comment (Lance52259) [inline openstack/apigw/dedicated/v2/whitelist/requests.go:46]:
```suggestion
// List is a method used to query the whitelists of endpoint service with given parameters.
```

File: openstack/apigw/dedicated/v2/whitelist/urls.go [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package whitelist
+
+import "github.com/chnsz/golangsdk"
+
+func listWhiteListURL(c *golangsdk.ServiceClient, instanceId string) string {
```
Comment (Lance52259) [inline openstack/apigw/dedicated/v2/whitelist/urls.go:5]:
```suggestion
func listURL(c *golangsdk.ServiceClient, instanceId string) string {
```

File: openstack/apigw/dedicated/v2/whitelist/requests.go [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+	"github.com/chnsz/golangsdk/pagination"
+)
+
+// CreateOpts is the structure that used to add whitelists for endpoint service.
+type CreateOpts struct {
+	// The Dedicated instance ID.
+	InstanceId string `json:"-" required:"true"`
+	// List of whitelists to add to endpoint service.
+	Permissions []string `json:"permissions" required:"true"`
+}
```
Comment (Lance52259) [inline openstack/apigw/dedicated/v2/whitelist/requests.go:14]:
```suggestion
// BatchOpts is the structure that used to add or remove whitelists for endpoint service.
type BatchOpts struct {
	// The Dedicated instance ID.
	InstanceId string `json:"-" required:"true"`
	// List of whitelists to add to endpoint service.
	Permissions []string `json:"permissions" required:"true"`
}
```
This structure is not just used for `Create()` method.

File: openstack/apigw/dedicated/v2/whitelist/requests.go [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	var r createResp
+	_, err = c.Post(deleteURL(c, opts.InstanceId), b, &r, &golangsdk.RequestOpts{
+		MoreHeaders: requestOpts.MoreHeaders,
+	})
+	return r.Permissions, err
+}
```
Comment (Lance52259) [inline openstack/apigw/dedicated/v2/whitelist/requests.go:78]:
```suggestion
// Delete is a method used to batch delete the whitelists of endpoint service using given parameters.
func Delete(c *golangsdk.ServiceClient, opts BatchOpts) error {
	b, err := golangsdk.BuildRequestBody(opts, "")
	if err != nil {
		return nil, err
	}

	_, err = c.Post(deleteURL(c, opts.InstanceId), b, nil, &golangsdk.RequestOpts{
		MoreHeaders: requestOpts.MoreHeaders,
	})
	return err
}
```

File: openstack/apigw/dedicated/v2/whitelist/requests.go (hunk 1 above)
Comment (jinyangyang222) [reply openstack/apigw/dedicated/v2/whitelist/requests.go]: fixed.

File: openstack/apigw/dedicated/v2/whitelist/urls.go (hunk 3 above)
Comment (jinyangyang222) [reply openstack/apigw/dedicated/v2/whitelist/urls.go]: fixed.

File: openstack/apigw/dedicated/v2/whitelist/requests.go (hunk 4 above)
Comment (jinyangyang222) [reply openstack/apigw/dedicated/v2/whitelist/requests.go]: fixed.

File: openstack/apigw/dedicated/v2/whitelist/requests.go (hunk 5 above)
Comment (jinyangyang222) [reply openstack/apigw/dedicated/v2/whitelist/requests.go]: fixed.

File: openstack/apigw/dedicated/v2/whitelist/requests.go (hunk 2 above)
Comment (jinyangyang222) [reply openstack/apigw/dedicated/v2/whitelist/requests.go]: fixed.

**Pull request merged**